### PR TITLE
feat: add `defaultTmId` and `defaultGlossaryId` support

### LINKED ISSUE #58: Projects API: `defaultTmId`, `defaultGlossaryId` support
The "Add Project", "Edit Project" API endpoint now supports the `defaultTmId` and `defaultGlossaryId` parameters. This allows you to set the default Translation Memory and Glossary for a project.

**Things to do:**

- Add new parameters to the Add Project and Edit Project methods.
- The Get Project response now also includes the default TM and Glossary IDs. It should be updated to reflect this change.

**References:**

- [Add Project](https://support.crowdin.com/developer/api/v2/#tag/Projects/operation/api.projects.post)
- [Edit Project](https://support.crowdin.com/developer/api/v2/#tag/Projects/operation/api.projects.patch)
- [Get Project](https://support.crowdin.com/developer/api/v2/#tag/Projects/operation/api.projects.get)

### PR DESCRIPTION
**Pull request by jameskim0987**
Resolves #58 

Changes made in this PR:
1. add `DefaultTMID` and `DefaultGlossaryID` in `ProjectsAddRequest`
2. add `DefaultTMID` and `DefaultGlossaryID` in the respective testcases
3. add `ProjectAiPreTranslate` and `ProjectAiPrompt` in `ProjectsAddRequest`
4. add `ProjectAiPreTranslate` and `ProjectAiPrompt` in the respective testcases
5. ~~add `FIXME` and `TODO` comments. Please review.~~ handled through conservations below

### CODE REVIEW

File: crowdin/model/projects.go [hunk 1]
```diff
@@ -243,6 +243,13 @@ type ProjectsAddRequest struct {
 	TMContextType  string                 `json:"tmContextType,omitempty"`
 	TMPreTranslate *ProjectTMPreTranslate `json:"tmPreTranslate,omitempty"`
 	MTPreTranslate *ProjectMTPreTranslate `json:"mtPreTranslate,omitempty"`
+	// FIXME: missing fields for aiPreTranslate and assistActionAiPromptId ( TODO: create new issue )
+	// Translation Memory ID.
+	// Default: null
```
Comment (andrii-bodnar) [inline crowdin/model/projects.go:248]: I would suggest adding these properties to the current PR and renaming it to a more generic name

File: crowdin/model/projects.go [hunk 2]
```diff
@@ -61,8 +61,8 @@ type (
 		LanguageMapping                 map[string]LanguageMapping `json:"languageMapping,omitempty"`
 		DelayedWorkflowStart            bool                       `json:"delayedTranslations,omitempty"`
 		NotificationSettings            *NotificationSettings      `json:"notificationSettings,omitempty"`
-		DefaultTMID                     int                        `json:"defaultTmId,omitempty"`
-		DefaultGlossaryID               int                        `json:"defaultGlossaryId,omitempty"`
+		DefaultTMID                     int                        `json:"defaultTmId,omitempty"`       // FIXME: shouldn't it be *int? looks like according to the API doc, this field is nullable
```
[the remark this answers is not in the document]
Comment (jameskim0987) [reply crowdin/model/projects.go:64]: I'm having trouble finding it but help me understand.
From this part given that default is `null`, shouldn't it be nullable?
Screen Shot 2024-10-01 at 10 48 50 AM

File: crowdin/model/projects.go (hunk 1 above)
Comment (jameskim0987) [reply crowdin/model/projects.go]: Added those properties and also added in the test `TestProjectsService_Add`

File: crowdin/model/projects.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jameskim0987) [reply crowdin/model/projects.go]: I see, thanks for the clarification! Removed the `FIXME` comments

**Comment by andrii-bodnar** [comment]
@jameskim0987 thank you for the contribution!

**Pull request merged**